### PR TITLE
Fix Algorithms Dependency Declaration

### PR DESCRIPTION
**Pull request by dankinsoid**
Swift Algorithms was used in Leaf but never explicitly declared as a dependency on the target. That resulted in compiler errors in some scenarios. The dependency is now correctly declared.

### CODE REVIEW

**Comment by 0xTim** [comment]
What version of Swift are you using etc? All CI is working and no one else seems to have reported any errors.

Regardless, we should fix the underlying error but I'd prefer to keep Swift Algorithms. Since it's used elsewhere in the Vapor stack you'll already have this dependency fetched and compiled

**Comment by dankinsoid** [comment]
@0xTim  I use swift 5.7
It compiles fine as a package, but doesn't compile as a dependency of another package (Vapor project in my case), I don't know why, but there is a mistake in the package file in my opinion
I change PR to save Algorithms, but fixed package file

**Comment by 0xTim** [comment]
Original comment:

My project build failed with 'No such module 'Alghorithms'' error. This is probably due to the fact that the Algorithms were not added to the target "Leaf" dependencies in the Project file
```swift
 .target(name: "Leaf", dependencies: [
            .product(name: "LeafKit", package: "leaf-kit"),
            .product(name: "Vapor", package: "vapor"),
        ]),
```
But as I see in all the code, Algorithms were used only once and not very justified, so I think it's better to remove it until you really need it

File: Sources/Leaf/LeafEncoder.swift [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import LeafKit
-import Algorithms
```
Comment (0xTim) [inline Sources/Leaf/LeafEncoder.swift:2]: We need to re-add this

**Review by 0xTim — CHANGES_REQUESTED** [review]

File: Sources/Leaf/LeafEncoder.swift [hunk 2]
```diff
@@ -79,7 +79,7 @@ extension LeafEncoder {
         }
         
         convenience init(from encoder: EncoderImpl, withKey key: CodingKey?) {
-            self.init(userInfo: encoder.userInfo, codingPath: encoder.codingPath + [key].compacted())
+            self.init(userInfo: encoder.userInfo, codingPath: encoder.codingPath + [key].compactMap { $0 })
```
Comment (0xTim) [inline Sources/Leaf/LeafEncoder.swift:82]: Revert
```suggestion
            self.init(userInfo: encoder.userInfo, codingPath: encoder.codingPath + [key].compacted())
```

**Review by 0xTim — APPROVED** [review]

**Pull request merged**